index.js

```diff
@@ -59,234 +48,161 @@ Fs.readdirSync(Path.join(__dirname, 'helpers')).forEach(function (file) {
+    * paper expects to be passed an assembler to gather all the templates required to render
+    * the top level template.
+    *
+    * @param {Object} siteSettings - Site settings
+    * @param {Object} themeSettings - Theme settings (configuration)
+    * @param {Object} assembler - Assembler with getTemplates and getTranslations methods.
+    * @param {String} rendererType - One of ['handlebars', 'handlebars-v4']
+    * @param {assemblerGetTemplates} assembler.getTemplates - Method to assemble templates
+    * @param {assemblerGetTranslations} assembler.getTranslations - Method to assemble translations
+    */
+    constructor(siteSettings, themeSettings, assembler, rendererType) {
+        this.siteSettings = siteSettings || {};
         this.themeSettings = themeSettings || {};
         this.assembler = assembler || {};
-        this.contentServiceContext = {};
 
-        helpers.forEach(helper => helper(this));
+        // Build renderer based on type
+        switch(rendererType) {
+        case 'handlebars-v4':
+            this.renderer = new HandlebarsRenderer(this.siteSettings, this.themeSettings, 'v4');
+            break;
+        case 'handlebars':
+        default:
+            this.renderer = new HandlebarsRenderer(this.siteSettings, this.themeSettings, 'v3');
+            break;
+        }
     }
 
     /**
-     * Renders a string with the given context
-     * @param  {String} string
-     * @param  {Object} context
-     */
-    renderString(string, context) {
-        return this.handlebars.compile(string)(context);
-    }
-
-    loadTheme(paths, acceptLanguage, done) {
+     * Use the assembler to fetch partials/templates, and translations, then load them
+     * into the renderer.
+     *
+     * @param {String|Array} paths A string or array of strings - the template path(s) to load.
+     * @param {String} acceptLanguage The accept-language header - used to select a locale.
+     * @param {Function} callback
+    */
+    loadTheme(paths, acceptLanguage, callback) {
         if (!_.isArray(paths)) {
             paths = paths ? [paths] : [];
         }
 
         Async.parallel([
-            (next) => {
+            next => {
                 this.loadTranslations(acceptLanguage, next);
             },
-            (next) => {
+            next => {
                 Async.map(paths, this.loadTemplates.bind(this), next);
             }
-        ], done);
+        ], callback);
     }
 
     /**
-     * Load Partials/Templates
-     * @param  {Object}   templates
-     * @param  {Function} callback
+     * Use the assembler to fetch partials/templates, then load them
+     * into the renderer.
+     *
+     * @param {String} path The root template path to load. All dependencies will be loaded as well.
+     * @param {Function} callback
     */
     loadTemplates(path, callback) {
-        let processor = this.getTemplateProcessor();
+        let preProcessor = this.renderer.getPreProcessor();
```

Comment:
Can we move this call to the constructor as a variable so we don't make this call for each path in the list ?